### DISCUSSION TITLE
How to avoid decrease in image response rate when the unreal window is not active?

### DISCUSSION THREAD

**Discussion by skykim0609** [microsoft/AirSim#3245] [Support Q&A] [answered]
### What's your question?

Hello. I am currently using airsim in Computer vision mode with versions below, and the problem is when the simulation is running and Unreal Editor window is not currently active, the image response of the Simulator decreases significantly. I know this is some sort of way to save computer resource, but I am trying to subscribe to image topics published via airsim_node(in airsim_ros_pkgs) and it turns out that this affects the publish rate of the image topics.

When the unreal window is not the "primary active" window (Depth cameras "camera_info" topic):
![image](https://user-images.githubusercontent.com/45422505/103008686-de208f80-4578-11eb-80a0-4a44d2298f97.png)
The image topic is published in only 3Hz

 But when the unreal window is the "primary active" window(When I click mouse on the simulation window)
![image](https://user-images.githubusercontent.com/45422505/103008783-001a1200-4579-11eb-9096-38a6ec9834fe.png)

The image topic is published much faster(about 13Hz I guess)

The question is, is there any way to avoid this reduction in image response rate when the simulation window is not on the top?

### Include context on what you are trying to achieve

I am trying to run a SLAM simulation on airsim, so 3 Hz image publish rate is not enough.

#### Context details

Specifications : 
Airsim version : 1.3.1
Unreal Engine version : 4.24.3
OS : Ubuntu 18.04

### Include details of what you already did to find answers

I searched for the issues with same problems. I couldn't get any hint from the existing issues.(Perhaps I searched wrong...)
I modified my  settings.json to "ViewMode": "NoDisplay", and "ClockSpeed": 1, but neither of these helped.

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @skykim0609 , This mainly is an Unreal Engine 'feature'. 
Try the solution shown [here](https://answers.unrealengine.com/questions/145958/huge-framerate-drop-when-game-window-unfocused-454.html) and let us know if it solves the issue 😊

Edit: Thanks for asking this btw. I never thought that this being enabled would affect the actual simulation.
Also, the option is now under Edit->Editor Preferences->General->Performance->"Use Less CPU when in Background"
